### PR TITLE
Documentation for how to specify mag and color cuts for supplemented ASTs

### PR DESCRIPTION
**Pull request by cmurray-astro**
In the docstring to the supplement_ast() function in make_ast_input_list.py, I added descriptions and examples of how to set the correct parameters in the beast_settings.txt file for making magnitude and color cuts to supplemented ASTs. I will also add this to the template beast_settings.txt file (which is stored in beast_examples).

@galaxyumi and @kmcquinn please let me know what you think! Based on your feedback, I can make this more user-friendly and/or intuitive, especially if you run into issues/bugs.

### CODE REVIEW

File: beast/observationmodel/ast/make_ast_input_list.py [hunk 1]
```diff
@@ -408,10 +408,52 @@ def supplement_ast(
...
+        # specify that the ast_supplement mode should be on
+        ast_supplement = True
+
+        # initialize the dictionary of desired magnitude limits
+        ast_suppl_colorlimits = {}
+
+        # the color limits are defined by the first filter in the color (e.g, X for X-Y),
+        # and the input is a list including the second filter (e.g., Y for X-Y) and the
+        # color limit in magnitudes
+        ast_suppl_colorlimits["HST_WFC3_F475W"] = [["HST_WFC3_F814W",4]]
```
Comment (galaxyumi) [inline beast/observationmodel/ast/make_ast_input_list.py:452]: To be consistent with the above description, it should be ast_suppl_colorlimits["HST_WFC3_F475W"] = [["HST_WFC3_F814W",6]]
ast_suppl_colorlimits["HST_WFC3_F336W"] = [["HST_WFC3_F475W",5], ["HST_WFC3_F814W",4]]

File: beast/observationmodel/ast/make_ast_input_list.py [hunk 2]
```diff
@@ -408,10 +408,52 @@ def supplement_ast(
...
+        # the magntidue limits are defined by the filter and a list of the limits in magnitudes
+        ast_suppl_maglimits["HST_WFC3_F475W"] = [16,28]
+        ast_suppl_maglimits["HST_WFC3_F814W"] = [15,27]
+
+        # set the key word
+        ast_suppl_maglimit = ast_suppl_maglimits
 
     color_cut: dictionary (optional, default=None)
-        Ditionary of faint color limits for given filters
+        Dictionary of faint color limits for given filters.
```
Comment (galaxyumi) [inline beast/observationmodel/ast/make_ast_input_list.py:433]: How about 'red'?

File: docs/generating_asts.rst [hunk 3]
```diff
@@ -134,7 +134,56 @@ Parameters used by the supplementing AST method
...
+
+  .. code-block:: console
+
+    # initialize the dictionary of desired magnitude limits
+    ast_suppl_colorlimits = {}
+
+    # the color limits are defined by the first filter in the color (e.g, X for X-Y),
+    # and the input is a list including the second filter (e.g., Y for X-Y) and the
+    # color limit in magnitudes:
+    ast_suppl_colorlimits["HST_WFC3_F475W"] = [["HST_WFC3_F814W",4]]
```
Comment (galaxyumi) [inline docs/generating_asts.rst:182]: Same comment for the beast/observationmodel/ast/make_ast_input_list.py.
ast_suppl_colorlimits["HST_WFC3_F475W"] = [["HST_WFC3_F814W",6]]
ast_suppl_colorlimits["HST_WFC3_F336W"] = [["HST_WFC3_F475W",5], ["HST_WFC3_F814W",4]]

**Comment by cmurray-astro** [comment]
also added descriptions to the generating_asts docs page!

**Review by galaxyumi — CHANGES_REQUESTED** [review]
Once the typos are fixed, it seems all good to me!

**Comment by kmcquinn** [comment]
I think the documentation you sent is clear. But I'm having trouble intergrating that with existing documentation. For example, there is a detailed description of making AST lists here:
https://beast.readthedocs.io/en/latest/generating_asts.html 
But I can't reconcile it with the beast_settings.txt documentation here:
https://beast.readthedocs.io/en/latest/beast_setup.html#beast-setup-settings

Perhaps I'm just missing something?

**Comment by cmurray-astro** [comment]
@kmcquinn good point -- the beast_settings documentation (https://beast.readthedocs.io/en/latest/beast_setup.html#beast-setup-settings) looks like a summary of all the settable parameters, without much guidance on how and when to use them (and also looks out of date!). I can open an issue about this.

**Comment by kmcquinn** [comment]
I'm working on testing the code to run MATCH-optimized ASTs and I realized that I'm not clear on how to specify ast_suppl_colorlimits.  If, for example, you have 4 filters, how do you specify the colorlimits variable? Do you need to list all different filter combinations with color limits? Would you be able to post an example of this should be done? Thanks!

**Comment by cmurray-astro** [comment]
@kmcquinn currently the way it works is you have to specify whichever color limits you want to impose. this PR includes documentation but hasn't been merged yet, so if you check out the "files changed" you can see the examples. the color limit example is here: 

        For example, for a field observed with HST_WFC3_F336W, HST_WFC3_F475W,
        and HST_WFC3_F814W, to set a color range limit of HST_WFC3_F475W-HST_WFC3_F814W<6,
        HST_WFC3_F336W-HST_WFC3_F475W<5 and HST_WFC3_F336W-HST_WFC3_F814W<4, you need
        to set the following within the beast_settings file:

        # specify that the ast_supplement mode should be on
        ast_supplement = True

        # initialize the dictionary of desired magnitude limits
        ast_suppl_colorlimits = {}

        # the color limits are defined by the first filter in the color (e.g, X for X-Y),
        # and the input is a list including the second filter (e.g., Y for X-Y) and the
        # color limit in magnitudes
        ast_suppl_colorlimits["HST_WFC3_F475W"] = [["HST_WFC3_F814W",6]]
        ast_suppl_colorlimits["HST_WFC3_F336W"] = [["HST_WFC3_F475W",5], ["HST_WFC3_F814W",4]]

**Comment by karllark** [comment]
Is this PR ready for merging?

**Comment by cmurray-astro** [comment]
Yep!

**Comment by kmcquinn** [comment]
Thanks! That is really helpful.

**Pull request merged**